app/src/main/java/org/mifos/mobile/presenters/LoginPresenter.kt

```diff
@@ -0,0 +1,195 @@
+                                                        .developerMessage)
+                                    }
+                                }
+                            } catch (throwable: Throwable) {
+                                RxJavaPlugins.getErrorHandler()
+                            }
+                        }
+
+                        override fun onNext(user: User) {
+                            if (user != null) {
+                                val userName = user.username
+                                val userID = user.userId
+                                val authToken = Constants.BASIC +
+                                        user.base64EncodedAuthenticationKey
+                                saveAuthenticationTokenForSession(userName, userID, authToken)
+                                mvpView!!.onLoginSuccess(userName)
+                            } else {
+                                mvpView!!.hideProgress()
+                            }
+                        }
+                    })
+            )
+        }
+    }
+
+    /**
+     * This method fetching the Client, associated with current Access Token.
+     */
+    fun loadClient() {
+        checkViewAttached()
+        compositeDisposable.add(dataManager.clients
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribeOn(Schedulers.io())
+                .subscribeWith(object : DisposableObserver<Page<Client>?>() {
+                    override fun onComplete() {}
+                    override fun onError(e: Throwable) {
+                        mvpView!!.hideProgress()
+                        if ((e as HttpException).code() == 401) {
+                            mvpView!!.showMessage(context.getString(R.string.unauthorized_client))
+                        } else {
+                            mvpView!!.showMessage(context.getString(R.string.error_fetching_client))
+                        }
+                        preferencesHelper.clear()
+                        reInitializeService()
+                    }
+
+                    override fun onNext(clientPage: Page<Client>) {
+                        mvpView!!.hideProgress()
+                        if (clientPage.pageItems.isNotEmpty()) {
+                            val clientId = clientPage.pageItems[0]!!.id.toLong()
+                            preferencesHelper.clientId = clientId
+                            dataManager.clientId = clientId
+                            reInitializeService()
+                            mvpView!!.showPassCodeActivity()
+                        } else {
+                            mvpView!!.showMessage(context
+                                    .getString(R.string.error_client_not_found))
+                        }
+                    }
+                })
+        )
+    }
+
+    @SuppressLint("StringFormatInvalid", "StringFormatMatches")
```

Comment:
Why is SuppressLint needed here and I saw it in other places, please clarify? Previously in Java, we were not using it.

Reply:
The string resources used in the following lines, i.e., `error_validation_blank`,`error_validation_minimum_chars`, `username_minimum_length` `error_validation_cannot_contain_spaces` are not in a valid format and thus it is not accepted by `String.format()`. Mostly this is because the translations were not done properly, meaning that those strings were not translated to all the languages in the app

Reply:
I see you removed it.